conans/client/build/cmake.py

```diff
@@ -241,8 +243,12 @@ def _run(self, command):
         elif is_intel:
             if self.generator in ["Ninja", "Ninja Multi-Config",
                                   "NMake Makefiles", "NMake Makefiles JOM", "Unix Makefiles"]:
-                intel_compilervars_dict = tools.intel_compilervars_dict(self._conanfile, force=True)
-                context = _environment_add(intel_compilervars_dict, post=self._append_vcvars)
+                compiler_version = self._settings.get_safe("compiler.version")
```

Comment:
No need to support legacy integrations.

Reply:
Then, if we don't need to support legacy ones, shall I keep only the newest Intel versions? -> `2021.x.x`

Reply:
I just realized about this is our old-fashioned cmake file, so, perhaps, here we can keep the legacy one and adapt the new `conan.tools.cmake` to the new oneAPI one.